PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -387,8 +387,7 @@ SetupBootImage (
     if (LoaderPlatformInfo != NULL) {
       if (FeaturePcdGet (PcdMeasuredBootEnabled) && (LoaderPlatformInfo->LdrFeatures & FEATURE_MEASURED_BOOT)) {
         // Extend hash of the image into TPM.
-        TpmExtendPcrAndLogEvent (8, TPM_ALG_SHA256, LoadedImage->ImageHash,
-          EV_COMPACT_HASH, sizeof("LinuxLoaderPkg: OS Image"), (UINT8 *)"LinuxLoaderPkg: OS Image");
+        TpmExtendOsHash (LoadedImage->ImageHash, HASH_TYPE_SHA256);

```

Comment:
Currently HASH_TYPE_SHA256 is used. Will improve to use the hash type from container format.

Reply:
Instead of adding this new API here, I would recommend to file a separate patch and take care of complete change w.r.t OS hash.

Reply:
TpmExtendPcrAndLogEvent() would enforce to use TPM Hash Alg. Also this API do not check for Active PCR banks. 
New API TpmExtendOsHash() would take care of additional checks.